crates/wasmtime/src/runtime/store.rs

```diff
@@ -478,6 +492,13 @@ impl<T> Store<T> {
     /// [`Store::limiter`] configuration method.
     pub fn new(engine: &Engine, data: T) -> Self {
         let pkey = engine.allocator().next_available_pkey();
+
+        let gc_store = Some(allocate_gc_store(engine).expect(
+            "TODO FITZGEN: should we propagate an error here? huge change for \
+             Store::new to become fallible tho... alternatively, could delay \
+             until instantiation?",
+        ));
```

Comment:
Yeah I think we want to keep `Store` allocation infallible if we can, so I think it'd be best to defer fallible work to when it's required rather than needing the unwrap here.